projects/igniteui-angular/src/lib/grids/grid/grid.component.ts

```diff
@@ -491,6 +491,27 @@ export class IgxGridComponent extends IgxGridBaseDirective implements GridType,
     @ContentChild(IgxDragIndicatorIconDirective, { read: TemplateRef, static: false })
     public dragIndicatorIconTemplate: TemplateRef<any> = null;
 
+    /**
+     * The custom drag ghost, if any, that should be used when rendering the drag ghost
+     *
+     * ```typescript
+     * // Set in typescript
+     * const myCustomGhostTemplate: TemplateRef<any> = myComponent.customGhostTemplate;
+     * myComponent.dragGhostCustomTemplate = myCustomGhostTemplate;
+     * ```
+     * ```html
+     * <!-- Set in markup -->
+     *  <igx-grid #grid>
+     *      ...
+     *      <ng-template igxDragCustomGhost>
+     *          <div class="dragGhost">Custom drag ghost!</div>
+     *      </ng-template>
+     *  </igx-grid>
+     * ```
+     */
+    @ContentChild(IgxRowDragGhost , { read: TemplateRef, static: false })
+    public dragGhostCustomTemplate: TemplateRef<any> = null;
```

Comment:
This being in `grid.component.ts` won't be inherited by other grids. It should be in `grid-base.directive.ts`, so it can be inherited. Also, since the HGrid can have multiple of those - it's better if you use ContentChilderen so you can ignore descendants. See the `rowSelectorsTemplates`/`rowSelectorTemplate` in the base grid.

That'd mean you'll also have a `@ContentChildren` dragGhostCustomTemplates and the dragGhostCustomTemplate will turn into a getter to give you the first of the children. You'll also need an override in the HGrid, since the template actually end up under the Row Island **next to ** the grid, so something like:
```ts
    public get dragGhostCustomTemplate(): TemplateRef<any> {
        if (this.parentIsland) {
            return this.parentIsland.dragGhostCustomTemplate;
        }
        return super.dragGhostCustomTemplate;
    }
```